src/flasher_spi.c

```diff
@@ -537,6 +537,235 @@ NETX_CONSOLEAPP_RESULT_T spi_flash(const FLASHER_SPI_FLASH_T *ptFlashDescription
 
 /*-----------------------------------*/
 
+/**
+ * \brief Check if a memory sector is empty (all 0xFF)
+ * 
+ * \param sectorSizeBytes  The size of the sector in byte
+ * \param sectorBuffer     Pointer to the first element in the sector buffer
+ * \return int             true (1==1) if empty, else false
+ */
+static int sectorIsEmpty(unsigned long sectorSizeBytes, unsigned char* sectorBuffer)
+{
+	unsigned char theByte = 0xFF;
+	for(unsigned long byteNr = 0; byteNr < sectorSizeBytes; byteNr++)
+	{
+		theByte = theByte & sectorBuffer[byteNr];
+	}
+	if(theByte == 0xFF)
+	{
+		return 1==1;
+	}
+
+	return 0==1;
+}
+
+/**
+ * @brief Sets the value of a single bit inside a bitmap
+ * 
+ * @param value   The value to set (0/1)
+ * @param bitPos  The position of the bit to set
+ * @param bitmap  Pointer to the bitmap which will be manipulated
+ */
+static void bitmapWriteBit(const unsigned int value, const unsigned long bitPos, unsigned int* bitmap)
+{
+	const unsigned int bitPerMapEntry = sizeof(unsigned int)*8;
+	unsigned int entry = bitPos/bitPerMapEntry;
+	unsigned int bitShift = bitPos%bitPerMapEntry;
+	if(value >= 1)
+	{
+		bitmap[entry] = bitmap[entry] | (1U << bitShift);
+	}
+	else
+	{
+		bitmap[entry] = bitmap[entry] & ~(1U << bitShift);
+	}
+	
+}
+
+/**
+ * @brief Reads the value of a specific bit inside a bitmap
+ * 
+ * @param bitPos  Position of the bit to read
+ * @param bitmap  Pointer to the bitmap which will be read
+ * @return int    Value of the requested bit (0/1)
+ */
+static int bitmapReadBit(unsigned long bitPos, unsigned int* bitmap)
+{
+	const unsigned int bitPerMapEntry = sizeof(unsigned int)*8;
+	unsigned int entry = bitPos/bitPerMapEntry;
+	unsigned int bitShift = bitPos%bitPerMapEntry;
+	return (bitmap[entry] >> bitShift) & 0x01;
+}
+
+
+/**
+ * @brief Erase a section of SPI memory using variably sized erase commands
+ * 
+ * @attention This function requires the list of erase operations pointed to by ptFlashDescription->tSpiErase
+ *            to be in sorted order with the smallest erase operation being at element 0
+ * 
+ * @param ptFlashDescription [in]  Device information returned by spi_detect.
+ * @param ulStartAdr         [in]  Start offset of the first erase block to be erased.
+ * @param ulEndAdr           [in]  End offset of the last erase block to be erased (offset of the last byte + 1).
+ * 
+ * @return
+ * - NETX_CONSOLEAPP_RESULT_OK: success, the memory has been erased.
+ * - NETX_CONSOLEAPP_RESULT_ERROR: An error has occurred.
+ */
+NETX_CONSOLEAPP_RESULT_T spi_smart_erase(const FLASHER_SPI_FLASH_T *ptFlashDescription, const unsigned long ulStartAdr, const unsigned long ulEndAdr)
+{
+	/* Be Pessimistic */
+	NETX_CONSOLEAPP_RESULT_T retVal = NETX_CONSOLEAPP_RESULT_ERROR;
+	const SPIFLASH_ATTRIBUTES_T attributes = ptFlashDescription->tAttributes;
+	const FLASHER_SPI_ERASE_T *eraseTypes = ptFlashDescription->tSpiErase;
+	
+	/* Nr of valid entries */
+	unsigned int nrEraseOps = 0;
```

Comment:
You could just store the number of valid erase operations when detecting them in the sdfp code.